src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,104 @@
+import { SyntheticEvent, forwardRef, useRef } from "react";
+import type { AriaToastProps } from "@react-aria/toast";
+import type { ToastState, QueuedToast } from "@react-stately/toast";
+import { useToast } from "@react-aria/toast";
+import type { CustomToastContent } from "./ToastProvider";
+
+import Button from "../Button/Button";
+import { st, classes } from "./toast.st.css";
+
+interface ToastProps<T> extends AriaToastProps<T> {
+  className?: string;
+  state: ToastState<T>;
+  toast: QueuedToast<T>;
+  icon: React.ReactNode;
+}
+
+function Toast({ state, ...props }: ToastProps<CustomToastContent>) {
+  const ref = useRef(null);
+
+  const { toastProps, titleProps, closeButtonProps } = useToast(
+    props,
+    state,
+    ref
+  );
+
+  const { icon } = props;
+
+  const {
+    content: {
+      title,
+      actionLabel = "",
+      shouldCloseOnAction = false,
+      onAction,
+      shouldShowIcon = true,
+    },
+    priority = 0,
+    animation,
+    key,
+  } = props.toast;
+
+  const priorities = ["info", "success", "warning", "error"] as const;
+
+  const priorityName = priorities[priority || 0];
+
+  const withOrWithoutCloseButtonProps = shouldCloseOnAction
+    ? closeButtonProps
+    : {};
+
+  if ((actionLabel && !onAction) || (!actionLabel && onAction)) {
+    throw new Error(
+      "Toast: actionLabel and onAction must be both present or both absent."
+    );
+  }
+
+  const onActionHandler = (
+    e: SyntheticEvent,
+    state: ToastState<CustomToastContent>
+  ) => {
+    if (onAction) {
+      onAction(e, state);
+    }
+  };
+
+  return (
+    <div
```

Comment:
It would be good if you can add some data-testid attributes to the div element to make it easier to target in tests.

Reply:
Good idea. I added data-testids.

Reply:
https://github.com/action-is-hope/shelley/pull/99/commits/55747aaf9f71d0ca5041b595d9e691cc014d03e5